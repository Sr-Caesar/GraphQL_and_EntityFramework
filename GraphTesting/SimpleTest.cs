


using GraphQL_Exercice.GraphQLSchema.Mutations;
using GraphTesting;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Configuration;

namespace Testing
{
    public class SimpleTest
    {
        [Fact]
        public async Task SchemaChangeTest()//CodeFirst or Anotation Based with schema as an artifact
        {
            var schema = await new ServiceCollection()
            .AddGraphQLServer()
            .AddQueryType<Query>()
            .BuildSchemaAsync();

            //var schema = await TestServices.Executor.GetSchemaAsync(default);

            schema.ToString().MatchSnapshot();
        }

        [Fact]//TODO https://www.youtube.com/watch?v=Nf7nX2H_iiM
        public async Task GetCompaniesTest()
        {
            var testBuilder = WebApplication.CreateBuilder(new string[] { });
            string myQuery = @"{
                  companies{
                    id
                    name
                    dateOfFundation
                  }
                }";
            var result = await new ServiceCollection()
            .AddDbContext<ExerciceData.Context.AppDbContext>(option =>
            {
                option.UseSqlServer(testBuilder.Configuration.GetConnectionString("DefaultConnection"));
            })
            .AddScoped<CompanyRepository>()
            .AddGraphQLServer()
            .AddQueryType<Query>()
            .ExecuteRequestAsync(myQuery);

            result.ToJson().MatchSnapshot();
        }
    }
}
