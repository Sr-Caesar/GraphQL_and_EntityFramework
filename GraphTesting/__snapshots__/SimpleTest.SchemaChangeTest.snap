schema {
  query: Query
}

type CompanyModel {
  id: Int!
  name: String!
  dateOfFundation: DateTime!
  drivers: [DriverModel!]
  tirs: [TirModel!]
}

type DriverModel {
  id: Int!
  name: String!
  surname: String!
  age: Int!
  tir: TirModel!
  tirId: Int!
  company: CompanyModel!
  companyId: Int!
}

type Query {
  companies: [CompanyModel!]!
}

type TirModel {
  id: Int!
  model: String!
  company: CompanyModel!
  companyId: Int!
  driverId: Int!
  driver: DriverModel!
}

"The `@specifiedBy` directive is used within the type system definition language to provide a URL for specifying the behavior of custom scalar definitions."
directive @specifiedBy("The specifiedBy URL points to a human-readable specification. This field will only read a result for scalar types." url: String!) on SCALAR

"The `DateTime` scalar represents an ISO-8601 compliant date time type."
scalar DateTime @specifiedBy(url: "https:\/\/www.graphql-scalars.com\/date-time")
